src/driver/amdxdna/amdxdna_drv.c

```diff
@@ -388,8 +423,40 @@ static int amdxdna_pmops_resume(struct device *dev)
 	return 0;
 }
 
+static int amdxdna_rpmops_suspend(struct device *dev)
+{
+	struct amdxdna_dev *xdna = pci_get_drvdata(to_pci_dev(dev));
+	int ret;
+
+	mutex_lock(&xdna->dev_lock);
+	if (!list_empty(&xdna->client_list)) {
+		XDNA_WARN(xdna, "can't suspend due to alive clients");
+		mutex_unlock(&xdna->dev_lock);
+		return -EBUSY;
+	}
```

Comment:
```suggestion
        WARN_ONCE(!list_empty(&xdna->client_list), "Can't suspend due to alive clients");
```
Based on your design, only after 5 seconds the last client was closed, this rumtime suspend will be called.
Thus here should be a WARN_ON(). If this happen, it is a driver bug that we must fix. No need to return EBUSY in this place.

Reply:
use WARN_ON()